src/main/java/com/ldtteam/blockui/AtlasManager.java

```diff
@@ -0,0 +1,121 @@
+package com.ldtteam.blockui;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiSpriteManager;
+import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.resources.TextureAtlasHolder;
+import net.minecraft.client.resources.metadata.gui.GuiMetadataSection;
+import net.minecraft.client.resources.metadata.gui.GuiSpriteScaling;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.neoforged.neoforge.client.event.RegisterClientReloadListenersEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Splits global vanilla gui atlas on per mod-id basis. Requires atlas definition in atlases/<mod_id>_gui.json. "directory" sources in
+ * the atlas definition must have unique path (ideally contain mod_id) because they join across all mod ids (blame mojang).
+ */
+public class AtlasManager
```

Comment:
We won't have to create a minecol atlas, right?

Reply:
It's automatic, read pr doc, cost u one JSON and file moves and XML fixes

Reply:
At least for outofjar loading we need "no atlast" loading. Can't we just use no atlast loading for our other images then too? =D

Reply:
I have never said we NEED atlas :D it's better to have it because of additional functionalities that only works for textures bound to atlas, but old ways still work.
Also technically speaking it might be possible to have oufofjar atlas
Really don't freak out from atlases, it's fully optional, and you can only benefit from them

Reply:
phew